pkg/operator/controller/canary/controller.go

```diff
@@ -129,9 +292,11 @@ func (r *reconciler) rotateRouteEndpoint(service *corev1.Service, current *route
 		return nil, fmt.Errorf("failed to rotate route port: %v", err)
 	}
 
-	_, err = r.updateCanaryRoute(current, updated)
+	changed, err := r.updateCanaryRoute(current, updated)
 	if err != nil {
 		return current, err
+	} else if !changed {
+		return current, fmt.Errorf("Expected canary route to updated: No relevant changes detected")
```

Comment:
Small typo:
"Expected canary route to updated: No relevant changes detected" -> 
"Expected canary route to be updated: No relevant changes detected"

Reply:
Fixed. Thanks!